### PR TITLE
WIP: Core refactor tests

### PR DESCRIPTION
**Pull request by nnance**
Refactor unit tests to be shared with express and hapi.  @helfer

### CODE REVIEW

**Comment by helfer** [comment]
I think this is a great start! We can probably have a `createApp` function that takes the options as first argument, and a string ('express', 'hapi', 'koa', etc. ) as a second argument. Ideally we'd be able to import the test suite and then tell it with which server to run the tests. We might even want to include express-graphql itself in those tests, so we can create a table that shows what features we support that they don't, and vice-versa. What do you think?

**Comment by nnance** [comment]
@helfer I haven't included your recommend changes above because I am not clear on what your trying to accomplish.  I think I took a different approach than you were suggestions in the fact that I organized it in way where the integration test suite is generic and is imported and ran by each specific integration module.  This made since to me because each integration has specific test it might need to execute and then run the generic suite.  Anyway, this is ready for final review.

File: src/integrations/integrations.test.ts [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+  excludeParser?: boolean;
+  apolloOptions?: ApolloOptions | {(): ApolloOptions | Promise<{}>};
+  graphiqlOptions?: GraphiQL.GraphiQLData;
+}
+
+export interface CreateAppFunc {
+    (options?: CreateAppOptions): void;
+}
+
+export default (createApp: CreateAppFunc) => {
```
Comment (helfer) [inline src/integrations/integrations.test.ts]: I would add a string parameter here to describe the tests, for example "Express" or "HAPI"

**Comment by helfer** [comment]
@nnance: I think this does basically what I meant for the importing part. I would add that string though so that can be printed in the test descriptions (unless we want to add another description around it, which would be fine, too, I guess.

At some point we should run express-graphql through our test suite as well, so we can see where our server does things differently than express-graphql.

**Comment by helfer** [comment]
This is great, btw! Have you run these tests for HAPI yet?

**Comment by nnance** [comment]
I agree with the string or at least someone to identify the different integrations.  I will try both ways.  I have another branch for my hapi work that I am keeping in sync with this branch and core branch that I have implemented the hapi test.  They all fail right now which isn't what I suspected.  but I wasn't going to include that work in this PR.

**Pull request merged**